addons/website_slides/static/src/js/slides_course_join.js

```diff
@@ -116,10 +127,18 @@ var CourseJoinWidget = publicWidget.Widget.extend({
                 self.afterJoin();
             } else {
                 if (data.error === 'public_user') {
-                    const message = data.error_signup_allowed ?
-                        _t('Please <a href="/web/login?redirect=%s">login</a> or <a href="/web/signup?redirect=%s">create an account</a> to join this course') :
-                        _t('Please <a href="/web/login?redirect=%s">login</a> to join this course');
-                    self._popoverAlert(self.$el, sprintf(message, encodeURIComponent(document.URL), encodeURIComponent(document.URL)));
+                    let message;
+                    if (self.invitePreview) {
+                        message = self.isPartnerWithoutUser ?
+                            _t('Please <a href="/slides/%s/identify?partner_id=%s&invite_hash=%s">create an account</a> to join this course') :
```

Comment:
So I'm trying to do both this + Markup. It feels complicated, and probably longer in practice :p, for instance this could become: 
```js
let message;
let tLogin = _t('login');
let tSignUp = _t('create an account');
if (self.invitePreview) {
    message = Markup(sprintf(
        _t('Please %s to join this course'),
        Markup`<a href="/slides/${channelId}/identify?invite_partner_id=${self.invitePartnerId}&invite_hash=${self.inviteHash}">${self.isPartnerWithoutUser ? tSignUp : tLogin}</a>`
    ));
    self._popoverAlert(self.$el, message);
}
```

Or we can simply add a Markup around the whole thing. (There may be useful to have encodeURIComponent too, as currently done.) What do you think?